subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc

```diff
@@ -1048,3 +1048,40 @@ Alternatively, for Kotlin implementations, the Kotlin compiler can be configured
 include::sample[dir="snippets/configurationCache/samConversion/groovy",files="build.gradle[tags=configure]"]
 include::sample[dir="snippets/configurationCache/samConversion/kotlin",files="build.gradle.kts[tags=configure]"]
 ====
+
+[[config_cache:not_yet_implemented:accessing_top_level_at_execution]]
+=== Accessing top-level methods and variables of a build script at execution time
+
+Currently, Gradle cannot run a task restored from the configuration cache if a task action calls a method defined directly in the build script.
+
+In case of the Groovy DSL, the action fails when running from the cache with an error message about method not being found.
+In the following snippet, the task `listFiles` uses such method:
+====
+include::sample[dir="snippets/configurationCache/topLevel/groovy",files="build.gradle[tags=not-supported]"]
+====
+
+Running the task the second time when the configuration cache is reused gives an error:
+----
+include::{snippetsPath}/configurationCache/topLevel/tests-groovy/notSupported.out[]
+----
+
+The workaround is to convert the method to the static method of a proper class:
```

Comment:
```suggestion
To prevent the task from failing when loaded from the configuration cache, convert the referenced top-level method to a static method within a class:
```
Suggestion: Switch wording from "the workaround" to "to prevent the task from failing when loaded from the configuration cache" to make the resolution more explicit. Minor wording changes for clarity and readability in the second clause.